File: Gestor_canciones2/src/gestorAplicacion/elementosLibreria/Cancion.java
package gestorAplicacion.elementosLibreria;

import java.util.ArrayList;
import java.io.Serializable;
import gestorAplicacion.personas.Artista;

public class Cancion  implements Serializable {
	
	private static final long serialVersionUID = 1L;
	
    public String nombre;
    private int duracion;
    private Artista artista;
    private Genero genero;
    private static int id_cancion = 1;
	private int id_realC;
    public static ArrayList<Cancion> cancionesCreadas= new ArrayList<Cancion>(); //atributo estatico para buscar en todas las canciones creadas
    
    public Cancion() {
    	
    }
    public Cancion(String nombre, int duracion, Artista artista, Genero genero){
        this.nombre = nombre;
        this.duracion = duracion;
        this.artista = artista;
        this.genero = genero;
        artista.getCanciones().add(this);
		id_realC = id_cancion;
		id_cancion++;
        Cancion.cancionesCreadas.add(this);
    }
    
	public void setNombre(String nombre) {
        this.nombre = nombre;
    }
    public String getNombre() {
        return nombre;
    }
    public void setDuracion(int duracion) {
        this.duracion = duracion;
    }
    public int getDuracion() {
        return duracion;
    }
    public void setArtista(Artista artista) {
        this.artista = artista;
    }
    public Artista getArtista() {
        return artista;
    }
    public void setGenero(Genero genero) {
        this.genero = genero;
    }
    public Genero getGenero() {
        return genero;
    }
    public void setId_cancion(int id_cancion) {
        this.id_realC = id_cancion;
    }
    public int getId_cancion() {
        return id_realC;
    }
    public static ArrayList<Cancion> getCancionesCreadas() {
		return cancionesCreadas;
	}

	public static void setCancionesCreadas(ArrayList<Cancion> cancionesCreadas) {
		Cancion.cancionesCreadas = cancionesCreadas;
	}

	public String play(){
        return "reproduciendo"+ this.nombre;
    }

    public String pause(){
        return "la canción" + this.nombre + "ha sido pausada";
    }
}

File: Phyton/Gestor_canciones/src/gestorAplicacion/elementosLibreria/Cancion.py
from gestorAplicacion.personas.Artista import Artista

class Cancion:
    id_cancion = 0
    cancionesCreadas = []
    def __init__(self, nombre, duracion, artista, genero):
        self.nombre = nombre
        self._duracion = duracion
        self._artista = artista
        self._genero = genero
        self._artista.getCanciones().append(self)
        Cancion.id_cancion += 1
        self._id_realC = Cancion.id_cancion
        Cancion.cancionesCreadas.append(self)
    def setNombre(self, nombre):
        self.nombre = nombre
    def getNombre(self):
        return self.nombre
    def setDuracion(self, duracion):
        self._duracion = duracion
    def getDuracion(self):
        return self._duracion
    def setArtista(self, artista):
        self._artista = artista
    def getArtista(self):
        return self._artista
    def setGenero(self, genero):
        self._genero = genero
    def getGenero(self):
        return self._genero
    def setId_cancion(self, id_cancion):
        self._id_realC = id_cancion
    def getId_cancion(self):
        return self._id_realC
    def play(self):
        return f"Reproduciendo: {self.nombre}"
    def pause(self):
        return "la cancion", self.nombre ,"ha sido pausada"